pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/adminoperation/ParticipantInitService.java

```diff
@@ -0,0 +1,178 @@
+package org.broadinstitute.dsm.service.adminoperation;
+
+import static org.broadinstitute.dsm.model.participant.data.FamilyMemberConstants.FAMILY_ID;
+import static org.broadinstitute.dsm.model.participant.data.FamilyMemberConstants.MEMBER_TYPE;
+import static org.broadinstitute.dsm.model.participant.data.FamilyMemberConstants.MEMBER_TYPE_SELF;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.google.gson.Gson;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.dto.ddp.participant.ParticipantData;
+import org.broadinstitute.dsm.exception.DSMBadRequestException;
+import org.broadinstitute.dsm.exception.DsmInternalError;
+import org.broadinstitute.dsm.exception.ESMissingParticipantDataException;
+import org.broadinstitute.dsm.model.elastic.Dsm;
+import org.broadinstitute.dsm.model.elastic.search.ElasticSearch;
+import org.broadinstitute.dsm.model.elastic.search.ElasticSearchParticipantDto;
+import org.broadinstitute.dsm.service.admin.AdminOperation;
+import org.broadinstitute.dsm.service.admin.AdminOperationRecord;
+import org.broadinstitute.dsm.service.participantdata.FamilyIdProvider;
+import org.broadinstitute.dsm.service.participantdata.RgpFamilyIdProvider;
+import org.broadinstitute.dsm.service.participantdata.RgpParticipantDataService;
+import org.broadinstitute.dsm.util.ElasticSearchUtil;
+
+@Slf4j
+public class ParticipantInitService implements AdminOperation {
+
+    private static final Gson gson = new Gson();
+    protected List<String> validRealms = List.of("rgp");
+    private DDPInstance ddpInstance;
+    private Map<String, Map<String, Object>> esParticipants;
+    private boolean isDryRun = true;
+
+    /**
+     * Validate input and retrieve participant data, during synchronous part of operation handling
+     *
+     * @param userId     ID of user performing operation
+     * @param realm      realm for fixup
+     * @param attributes unused
+     * @param payload    request body, if any, as ParticipantDataFixupRequest
+     */
+    public void initialize(String userId, String realm, Map<String, String> attributes, String payload) {
+        if (!validRealms.contains(realm.toLowerCase())) {
+            throw new DsmInternalError("Invalid realm for ParticipantInitService: " + realm);
+        }
+
+        ddpInstance = DDPInstance.getDDPInstance(realm);
+        if (ddpInstance == null) {
+            throw new DsmInternalError("Invalid realm: " + realm);
+        }
+
+        String esIndex = ddpInstance.getParticipantIndexES();
+        if (StringUtils.isEmpty(esIndex)) {
+            throw new DsmInternalError("No ES participant index for study " + realm);
+        }
+
+        // require dryRun since the user cannot specify a participant list
+        String dryRun = attributes.get("dryRun");
```

Comment:
Per the PR comments, I wanted the user to be specific about what they want to do, since this operation affects potentially all participants in a study. IOW, the user should definitely do a dry run first, so defaulting to no dry run is not what we want here.
We could also default to dry run = true, but at that point the user needs to know about the dry run flag anyway, so just requiring this seems more straightforward.
Does that make sense?

Reply:
I meant by default dryrun = false but like I said I am ok with enforcing